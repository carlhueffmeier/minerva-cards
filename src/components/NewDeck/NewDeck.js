import React, { Component } from 'react';
import { connect } from 'react-redux';
import { operations as deckOperations } from 'state/modules/decks';

export class NewDeck extends Component {
  state = {
    deckTitle: ``
  };

  handleChange(event) {
    this.setState({
      deckTitle: event.target.value
    });
  }

  handleSubmit(event) {
    const newDeckTitle = this.state.deckTitle.trim();
    this.props.addDeck(newDeckTitle);
    this.setState({ deckTitle: `` });
    event.preventDefault();
  }

  render() {
    const { deckTitle } = this.state;
    return (
      <form onSubmit={this.handleSubmit.bind(this)}>
        <input value={deckTitle} onChange={this.handleChange.bind(this)} />
        <button>Add</button>
      </form>
    );
  }
}

export default connect(null, { addDeck: deckOperations.addDeck })(NewDeck);
