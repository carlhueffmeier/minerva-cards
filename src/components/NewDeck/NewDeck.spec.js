import React from 'react';
import { shallow } from 'enzyme';
import { NewDeck } from './NewDeck';

describe('NewDeck', () => {
  const newDeck = shallow(<NewDeck />);

  it('renders properly', () => {
    expect(newDeck).toMatchSnapshot();
  });
});
