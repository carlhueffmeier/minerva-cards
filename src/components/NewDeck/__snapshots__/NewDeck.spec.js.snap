// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewDeck renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewDeck />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          onChange={[Function]}
          value=""
/>,
        <button>
          Add
</button>,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add",
        },
        "ref": null,
        "rendered": "Add",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            value=""
/>,
          <button>
            Add
</button>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add",
          },
          "ref": null,
          "rendered": "Add",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
